### ISSUE TITLE
GetHashCode throws system overflow exception in Win 10 x64

### ISSUE THREAD

**Issue by mytnyk** [AndresTraks/BulletSharpPInvoke#16]
All the places (I found 4 places) where you do have IntPtr casts to int naturally throws system overflow exception

```
public override int GetHashCode()
    {
        return _native.ToInt32();
    }
```

I solved this with the following change:

```
public override int GetHashCode()
    {
        return _native.GetHashCode();
    }
```

I have no idea why this does not reproduce for Win 7. Is there any special reason you do not use GetHashCode() of IntPtr but prefer ToInt32() ? If not, then I suggest to use GetHashCode().

**Comment by AndresTraks** [comment]
It seemed to be the fastest solution, but I did not consider the overflow case on 64-bit at the time. Don't know why it doesn't crash on Windows 7, maybe the address space is different and the pointer fits below 4 GB there?

Anyway, you are right, [IntPtr.GetHashCode](http://referencesource.microsoft.com/#mscorlib/system/intptr.cs,113) seems to do the right thing, so I changed it. There were only 3 places though (CollisionObject, CollisionShape, SoftBody)?

Thanks!
